randovania/games/samus_returns/generator/bootstrap.py

```diff
@@ -87,8 +97,17 @@ def assign_pool_results(self, rng: Random, patches: GamePatches, pool_results: P
             pickup for pickup in list(pool_results.to_place) if pickup.pickup_category is METROID_DNA_CATEGORY
         ]
 
+        if len(dna_to_assign) > len(locations):
+            raise InvalidConfiguration(
+                f"Has {len(dna_to_assign)} DNA in the pool, but only {len(locations)} valid locations."
+            )
+
         for dna, location in zip(dna_to_assign, locations, strict=False):
             pool_results.to_place.remove(dna)
             pool_results.assignment[location.pickup_index] = dna
 
         return super().assign_pool_results(rng, patches, pool_results)
+
+
+_boss_indices = [37, 139, 171]
+_stronger_metroid_indices = [177, 178, 181, 185, 186, 187, 188, 192, 193, 199, 200, 202, 205, 209]
```

Comment:
If it isn't required elsewhere in this module, just move it to the function `all_dna_locations` itself.

Reply:
Will do